src/riak_kv_qry_compiler.erl

```diff
@@ -436,7 +443,24 @@ col_index_and_type_of(Fields, ColumnName) ->
             {Position, Type}
     end.
 
-expand_where(Where, #key_v1{ast = PAST}, MaxSubQueries) ->
+%%
+-spec expand_where(filter(), #key_v1{}, integer()) ->
+        [where_props()] | {error, any()}.
+expand_where(Where, PartitionKey, MaxSubQueries) ->
+    case find_quanta_function_in_key(PartitionKey) of
+        [{[QField], QSize, QUnit}] ->
```

Comment:
This is somewhat fragile to only permitting a single quantization in the partitioning - which is fair for now.  Generally we may want to allow multiple quantization function in the partition key, e.g. if you wanted to split a high volume ingest stream by sharding an eventid to spread the load, or possible if you multiple timestamps like a created/ingested and wanted to quantize on both.